frontend/src/hooks/useFormatUtils.ts

```diff
@@ -0,0 +1,152 @@
+import { MouseEvent, useCallback } from "react";
+import { EditorState, Text, EditorSelection, Transaction } from "@codemirror/state";
+import { EditorView } from "@codemirror/view";
+import { indentWithTab } from "@codemirror/commands";
+import { Dispatch, SetStateAction } from "react";
+import { useSelector } from "react-redux";
+import { selectEditor } from "../store/editorSlice";
+
+export interface FormatBarState {
+	show: boolean;
+	position: { top: number; left: number };
+	selectedFormats: Set<FormatType>;
+}
+
+export enum FormatType {
+	BOLD = "bold",
+	ITALIC = "italic",
+	CODE = "code",
+	STRIKETHROUGH = "strikeThrough",
+}
+
+export const useFormatUtils = () => {
```

Comment:
I created customHook `useFormatUtils.ts `.  and extract format&formatBar's functions